src/kirby/components/list/directives/infinite-scroll.directive.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  Directive,
+  AfterViewInit,
+  OnDestroy,
+  HostListener,
+  EventEmitter,
+  Input,
+  Output,
+  ElementRef,
+  Inject,
+} from '@angular/core';
+import { Subject } from 'rxjs';
+import { debounceTime, takeUntil, filter } from 'rxjs/operators';
+
+import { WindowRef } from './../../shared/window-ref/window-ref.service';
+import { Scroll } from './scroll.model';
+
+@Directive({
+  selector: '[kirbyInfiniteScroll]',
+})
+export class InfiniteScrollDirective implements AfterViewInit, OnDestroy {
+  /**
+   * Event that will be triggered when user has scrolled to
+   * bottom of the element
+   */
+  @Output() public scrollEnd = new EventEmitter<void>();
+
+  /**
+   * If true then `scrollEnd` event should NOT be emitted
+   */
+  @Input() public disabled = false;
+
+  /**
+   * Emits a new value on element scroll event
+   */
+  public scroll$: Subject<Scroll> = new Subject<Scroll>();
```

Comment:
Question... do we prefix publicly-accessible properties / methods with the `public`-access modifier (it's implicitly applied)?

Reply:
I have removed them now. But maybe we should have a talk about what style Kirby should follow on this?